instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py

```diff
@@ -110,7 +110,7 @@ def _before_request():
         return
 
     environ = flask.request.environ
-    span_name = flask.request.endpoint or otel_wsgi.get_default_span_name(
+    span_name = flask.request.url_rule.rule or otel_wsgi.get_default_span_name(
```

Comment:
Not familiar enough with flask. Is url_rule always available? What if the rule is not defined and the request is a 404?

Reply:
It is not always available. If it's not, we default to the wsgi implementation which is HTTP + <method>

Reply:
Would this raise attribute error if `request.url_rule` is not always present? or is `url_rule` always present but `rule` can be None?

Reply:
I added a try catch for attribute error in my latest commit.